overlord/snapstate/snapmgr.go

```diff
@@ -439,27 +433,32 @@ func (m *SnapManager) ensureRefreshes() error {
 
 	tr := config.NewTransaction(m.state)
 
-	// allow disabling auto-refresh in the tests
-	if osutil.GetenvBool("SNAPD_DEBUG") {
-		var refreshDisabled bool
-		err := tr.Get("core", "refresh.disabled", &refreshDisabled)
-		if err != nil && !config.IsNoOption(err) {
-			return err
-		}
-		if refreshDisabled {
-			return nil
-		}
+	// already have a refresh timer
+	if m.nextRefresh != nil {
```

Comment:
I couldn't find logic that reload the next refresh when the value changes, and stops the old timer while doing that to ensure it doesn't get called out of turn. Does that exist and I missed, or do we still need it?